source/write/bulk-write.txt

```diff
@@ -340,6 +362,258 @@ subscribing to the observable and using the following methods:
+collection. It also sets the value of the ``manufacturer`` field to
+``"Premium Technologies"`` in all documents that have a ``category``
+value of ``"electronic"`` in the ``things`` collection.
+
+If multiple documents match the query filter specified in
+a ``ClientNamespacedUpdateOneModel`` instance, the operation updates the
+first result. You can specify a sort order in a `ClientUpdateOneOptions
+<{+core-api+}/com/mongodb/client/model/bulk/ClientUpdateOneOptions.html>`__
+instance to apply an order to matched documents before the driver
+performs the update operation, as shown in the following code:
+
+.. code-block:: scala
+
+   val options = ClientUpdateOneOptions
+     .clientUpdateOneOptions()
+     .sort(ascending("_id"))
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+The following example shows how to create models to replace
+existing documents in the ``db.people`` and ``db.things`` collections:
+
+.. literalinclude:: /includes/write/client-bulk-write.scala
+   :start-after: start-replace-models
+   :end-before: end-replace-models
+   :language: scala
+   :copyable:
+   :dedent:
+
+After this example runs successfully, the document that has an ``_id`` value of ``1`` 
+in the ``people`` collection is replaced with a new document. The document in 
+the ``things`` collection that has an ``_id`` value of ``1`` 
+is replaced with a new document.
+
+If multiple documents match the query filter specified in
+a ``ClientNamespacedReplaceOneModel`` instance, the operation replaces the
+first result. You can specify a sort order in a `ClientReplaceOneOptions
+<{+core-api+}/com/mongodb/client/model/bulk/ClientReplaceOneOptions.html>`__
+instance to apply an order to matched documents before the driver
+performs the replace operation, as shown in the following code:
+
+.. code-block:: scala
+
+   val options = ClientReplaceOneOptions
+     .clientReplaceOneOptions()
+     .sort(ascending("_id"))
+
+Perform the Bulk Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After you define a ``ClientNamespacedWriteModel`` instance for each
+operation you want to perform, pass a list of these instances to the
+client ``bulkWrite()`` method. By default, the method runs the
+operations in the order they're specified.
+
+The following example performs multiple write operations by using the
+``bulkWrite()`` method:
+
+.. io-code-block::
+   :copyable: 
+
+   .. input:: /includes/write/client-bulk-write.scala
+      :start-after: start-perform
+      :end-before: end-perform
+      :language: scala
+      :dedent:
+
+   .. output::
+      
+      AcknowledgedSummaryClientBulkWriteResult{insertedCount=1, matchedCount=1, ...}
+
+If any of the write operations fail, the driver raises a
+``ClientBulkWriteException`` and does not perform any further individual operations.
+``ClientBulkWriteException`` includes a ``BulkWriteError`` that can be accessed by using the
+``ClientBulkWriteException.getWriteErrors()`` method, which provides details of the
+individual failure.
```

Comment:
s: can there be multiple failures? if so, i think this should be 'individual failures.' if not, i would go with just 'the failure.'